### ISSUE TITLE
Different featureType attributes in different groups

### ISSUE THREAD

**Issue by Phil3175** [cf-convention/discuss#93]
Hi all,

I would like to create a netCDF file with various discrete sampling geometries (time series, trajectories, profiles...).

In the current version of CF, one can read:
 "all the collections in a CF file must be of the same feature type . (Future versions of CF may allow mixing of multiple feature types within a file.)"

I suppose it means that it is not allowed to use different featureType attributes in different netCDF groups? If not, I think it would be useful to add this possibility in future versions of CF.

Thanks,
Philippe

**Comment by erget** [comment]
Hi @Phil3175 , indeed, we have discussed this in the past and the general feeling was that it would be good to be able to use different feature types within the same file. There would be some work required to make this admissible, i.e. describing it in a way that maintains the internal consistency of the CF Conventions document and updating the compliance document. I believe the only thing preventing this from already being done is that nobody's had time to do it yet. So if you'd like to give it a shot, feel free to jump in!

**Comment by davidhassell** [comment]
I agree. This was discussed in a past CF workshop (2017 or 2019, I can't remember which)  and it was thought to be a good idea, but clearly one which needs a protagonist to carry out work that @erget outlines.

I think that there has never been any conceptual issues with allowing per-variable (or per-group) featureTypes, rather the restriction of one featureType per file was merely considered practical and covered the use cases at that time. So extending this to cater for today's use cases would be a good thing.

**Comment by JonathanGregory** [comment]
Dear Phil

I agree with you, Daniel and David that this would be a good extension. It requires `featureType` to be a variable and group attribute as well as global, but I don't think there's a conceptual problem. It was limited to global because that was a limitation of the only software available at the time.

Best wishes

Jonathan
